### DISCUSSION TITLE
HELP with debug API on Windows 11

### DISCUSSION THREAD

**Discussion by drossi-dg** [openkfw/TruBudget#1428] [General]
Hi Guys!

This is my situation: I could set up locally all docker images and run containers with `docker compose` on terminal. Then, I see the containers on **Docker Desktop** app and I can use the API, Frontend, etc.

The problem is that I need to debug the API and I follow de documentation but I can't debug (I'm using Visual Studio code on Windows 11)

For example, if I try to start only (not debug) API, y execute `npm run start` but API is down. Only I can start API with Docker Desktop.

Does anyone know how I can do it? Thank you!

**Comment by SamuelPull** [comment]
Hi drossi-dg, 

easy way to do this is to run the application and attach debugger to api server running in the docker compose container using `Docker: Attach to API Container`  debug configuration in VS Code.

**Reply by drossi-dg** [reply]
Hey Samuel, thank for your answer. I'm going to research about Docker: Attach to API Container debug configuration in VS Code.
